### ISSUE TITLE
'no-use-before-define' 'nofunc' should be enhanced.

### ISSUE THREAD

**Issue by omeryagmurlu** [eslint/eslint#5088]
Consider the following code.

```js
/* eslint no-use-before-define: [2, "nofunc"] */
'use strict';
class Elma {
	constructor() {}

	tellName() {
		return 'Elma';
	}

	sayHiWithFriends() {
		let friend = new Armut();
		console.log(this.tellName(), 'and', friend.tellName());
	}
}

class Armut {
	constructor() {}

	tellName() {
		return 'Armut';
	}

	sayHiWithFriends() {
		let friend = new Elma();
		console.log(this.tellName(), 'and', friend.tellName());
	}
}

let e = new Elma();
e.sayHiWithFriends();

let a = new Armut();
a.sayHiWithFriends();
```

outputs: 
```
Elma and Armut
Armut and Elma
```
as expected.

This request is not a duplicate of [#2874](https://github.com/eslint/eslint/issues/2874) and is not about hoisting or TDZ etc.

Classes depending on each other cause a false 'no-use-before-define'.

Eslint outputs:
```
  11:20  error  "Armut" was used before it was defined  no-use-before-define

✖ 1 problem (1 error, 0 warnings)

```

Eslint version: v1.10.3

**Comment by nzakas** [comment]
Duplicate of https://github.com/eslint/eslint/issues/3944. Fix is already on master.
